workflow/rocoto/tasks.py

```diff
@@ -176,6 +176,203 @@ def _get_forecast_hours(run, config, component='atmos') -> List[str]:
 
         return fhrs
 
+    @staticmethod
+    def get_job_groups(fhrs: List[int], ngroups: int, breakpoints: List[int] = None) -> List[dict]:
+        '''
+        Split forecast hours into a number of groups, obeying a list of pre-set breakpoints.
+
+        Takes a list of forecast hours and splits it into a number of groups while obeying
+        a list of pre-set breakpoints and recording which segment each belongs to.
+
+        Parameters
+        ----------
+        fhrs: List[int]
+                List of forecast hours to break into groups
+        ngroups: int
+                 Number of groups to split the forecast hours into
+        breakpoints: List[int]
+                     List of preset forecast hour break points to use (default: [])
+
+        Returns
+        -------
+        List[dict]: List of dicts, where each dict contains two keys:
+                    'fhrs': the forecast hours for that group
+                    'seg': the forecast segment (from the original breakpoint list)
+                           the group belong to
+        '''
+        if breakpoints is None:
+            breakpoints = []
+
+        num_segs = len(breakpoints) + 1
+        if num_segs > ngroups:
+            raise ValueError(f"Number of segments ({num_segs}) is greater than the number of groups ({ngroups}")
+
+        if ngroups > len(fhrs):
+            ngroups = len(fhrs)
+
+        # First, split at segment boundaries
+        fhrs_segs = [grp.tolist() for grp in np.array_split(fhrs, [fhrs.index(bpnt) + 1 for bpnt in breakpoints])]
+        seg_lens = [len(seg) for seg in fhrs_segs]
+
+        # Initialize each segment to be split into one job group
+        ngroups_segs = [1 for _ in range(0, len(fhrs_segs))]
+
+        # For remaining job groups, iteratively assign to the segment with the most
+        # hours per group
+        for _ in range(0, ngroups - len(fhrs_segs)):
+            current_lens = [size / weight for size, weight in zip(seg_lens, ngroups_segs)]
+            index_max = max(range(len(current_lens)), key=current_lens.__getitem__)
+            ngroups_segs[index_max] += 1
+
+        # Now that we know how many groups each forecast segment should be split into,
+        # Split them and flatten to a single list.
+        groups = []
+        for seg_num, (fhrs_seg, ngroups_seg) in enumerate(zip(fhrs_segs, ngroups_segs)):
+            [groups.append({'fhrs': grp.tolist(), 'seg': seg_num}) for grp in np.array_split(fhrs_seg, ngroups_seg)]
+
+        return groups
+
+    @staticmethod
+    def test_job_groups():
```

Comment:
Should we create and place the tests in `test_tasks.py`?  How are these tests executed?

Reply:
Right now, they are only executed in an interactive python interpreter by importing the class and then calling the test methods.

Instinctually I like the tests in the same file they are testing, but after doing some cursory research, it seems the preferred way is a separate file, so should probably go in that direction. I can either do it here, or in a future PR.

Reply:
Since tests were added (a good thing), I suggest placing them in the separate file such as `test_tasks.py`.  
It will take a little effort to make them part of a `pytest`, but that can be addressed in another PR.